tests/test_10_base.py

```diff
@@ -15,66 +11,51 @@ class BaseNameTest(Archive):
-            )
-        except ValueError as e:
-            self.fail(repr(e))
-        try:
-            self.assertEqual(
-                "xxx", BaseNameTest._guess("xxx", "xxx.ext1", fileobj)
-            )
-        except ValueError:
-            pass
-        else:
-            self.fail("guessing should has failed")
-
-
-class ArchiveFileTest(unittest.TestCase):
-    def _regular_tests(self, archive, fileobj, filename, text):
-        self.assertEqual(
-            archive.fileno(), fileobj.fileno(), "file no does not match!"
-        )
-        self.assertEqual(
-            archive.name, filename, "name attribute does not match!"
-        )
+def test_10_guess_basename():
+    fileobj = BytesIO(b"")
+    assert "xxx" == BaseNameTest._guess("mime2", "xxx", fileobj)
+    assert "xxx" == BaseNameTest._guess("mime1", "xxx.ext2", fileobj)
+    assert "xxx" == BaseNameTest._guess("mime2", "xxx.ext1", fileobj)
+    with pytest.raises(ValueError):
+        assert BaseNameTest._guess("xxx", "xxx.ext1", fileobj)
+
+
+def test_10_passing_file_object(tmp_path):
+    text = b"Hello World!\n"
+    with (tmp_path / "testfile").open("w+b") as fileobj:
+        fileobj.write(text)
+        fileobj.flush()
+        archive = ArchiveFile(fileobj=fileobj)
+        assert archive.fileno() == fileobj.fileno()
+        assert archive.name == fileobj.name
         archive.seek(0)
-        self.assertEqual(archive.read(), text, "file content does not match!")
-
-    def test_10_passing_file_object(self):
-        text = b"Hello World!\n"
-        with TemporaryFile() as fileobj:
-            fileobj.write(text)
-            fileobj.flush()
-            archive = ArchiveFile(fileobj=fileobj)
-            self._regular_tests(archive, fileobj, fileobj.name, text)
-
-    def test_20_passing_filename(self):
-        text = b"Hello World!\n"
-        with TemporaryFile() as fileobj:
-            fileobj.write(text)
-            fileobj.flush()
-            archive = ArchiveFile(fileobj=fileobj)
-            self._regular_tests(archive, fileobj, fileobj.name, text)
-
-    def test_30_closefd(self):
-        fileobj = BytesIO()
-        archive = ArchiveFile(fileobj=fileobj, closefd=False)
-        archive.close()
-        self.assertFalse(fileobj.closed)
-        archive = ArchiveFile(fileobj=fileobj, closefd=True)
-        archive.close()
-        self.assertTrue(fileobj.closed)
+        assert archive.read() == text
+
+
+@pytest.mark.xfail(
+    reason="changed to name=filename while pytestizing, but fails"
+)
+def test_20_passing_file_name(tmp_path):
+    text = b"Hello World!\n"
+    filename = tmp_path / "testfile"
+    with filename.open("w+b") as fileobj:
+        fileobj.write(text)
+        fileobj.flush()
+        archive = ArchiveFile(name=filename)
+        assert archive.fileno() == fileobj.fileno()
+        assert archive.name == filename
+        archive.seek(0)
+        assert archive.read() == text
```

Comment:
```suggestion
    with filename.open("w+b") as fileobj:
        fileobj.write(text)
    archive = ArchiveFile(name=filename)
    assert archive.name == filename
    assert archive.read() == text
```

The test is rubbish XD let me fix that